lib/presentation/bloc/bloc_impl.dart

```diff
@@ -0,0 +1,72 @@
+import 'dart:async';
+
+import '../../domain/entity/movie.dart';
+import '../../domain/use_cases/implementations/use_case_impl.dart';
+import '../../domain/use_cases/use_case_interface.dart';
+import 'bloc.dart';
+
+class BlocImpl extends Bloc {
+  final UseCases useCases;
+  final _moviesStream = StreamController<List<MovieEntity>>();
+  Map<int, String> genres = {};
+
+  @override
+  Stream<List<MovieEntity>> get movieStream => _moviesStream.stream;
+
+  @override
+  void dispose() {
+    //  _moviesStream.close();
+  }
+
+  BlocImpl({this.useCases = const UseCases()}) {
+    fetchGenres();
+  }
+
+  @override
+  void fetchPopularMovies() async {
+    EitherMovieAPI<List<MovieEntity>> movies =
```

Comment:
Review this class, made you own implementation

Reply:
addressed